### ISSUE TITLE
ImportError: huggingface-hub>=0.19.3,<1.0 is required for a normal functioning of this module, but found huggingface-hub==0.17.1

### ISSUE THREAD

**Issue by nlpkiddo-2001** [huggingface/huggingface_hub#2377]
Hi, I have installed huggingface-hub latest version 0.23.4, but I am getting the below error while running my code. I will share my environment details along with this. Please help me to fix this. Thanks In Advance.

```ImportError: huggingface-hub>=0.19.3,<1.0 is required for a normal functioning of this module, but found huggingface-hub==0.17.1```

**### Pip package List**

`absl-py                           1.3.0
aiohttp                           3.9.3
antlr4-python3-runtime            4.8
astunparse                        1.6.3
attrs                             17.3.0
beautifulsoup4                    4.11.1
bitarray                          2.6.2
bleach                            5.0.1
blis                              0.7.9
boto3                             1.26.0
botocore                          1.29.44
Bottleneck                        1.3.5
cachetools                        5.2.0
catalogue                         2.0.8
certifi                           2024.7.4
chardet                           3.0.4
charset-normalizer                2.1.1
click                             7.1.2
cloudpickle                       2.2.0
cmake                             3.30.0
colorama                          0.4.6
confection                        0.0.4
contourpy                         1.0.6
cryptography                      36.0.0
ctranslate2                       3.18.0
cupy-cuda11x                      12.2.0
cycler                            0.10.0
cymem                             2.0.7
Cython                            0.29.32
cytoolz                           0.12.0
dask                              2022.12.1
dill                              0.3.5
distributed                       2022.12.1
dlib                              19.17.0
dnspython                         2.2.1
easyocr                           1.6.2
eli5                              0.13.0
et-xmlfile                        1.1.0
face-recognition                  1.2.3
face-recognition-models           0.3.0
fairseq                           0.12.2
fastai                            1.0.60
fastapi                           0.103.2
fastprogress                      1.0.3
fastrlock                         0.8.2
featuretools                      0.5.1
filelock                          3.10.4
flatbuffers                       22.12.6
fonttools                         4.38.0
fsspec                            2023.5.0
future                            0.18.2
fvcore                            0.1.5.post20221221
gast                              0.4.0
gensim                            4.2.0
gibberish-detector                0.1.1
google-auth                       2.15.0
google-auth-oauthlib              0.4.6
google-pasta                      0.2.0
graphviz                          0.20.1
grpcio                            1.51.1
h5py                              3.6.0
hdbscan                           0.8.37
HeapDict                          1.0.1
hnswlib                           0.6.2
holidays                          0.9.9
html5lib                          0.9999999
huggingface-hub                   0.23.4
hydra-core                        1.0.7
hyperopt                          0.1.1
idna                              3.4
imageio                           2.23.0
imbalanced-learn                  0.8.1
importlib-metadata                6.0.0
imutils                           0.4.5
inflect                           6.0.1
install                           1.3.5
iopath                            0.1.9
Jinja2                            3.1.2
jmespath                          1.0.1
joblib                            1.2.0
keras                             2.10.0
Keras-Preprocessing               1.1.2
kiwisolver                        1.1.0
kmodes                            0.10.1
langcodes                         3.3.0
layoutparser                      0.3.4
libclang                          14.0.6
lime                              0.1.1.32
llvmlite                          0.39.0
lm-format-enforcer                0.10.1
locket                            1.0.0
lxml                              4.9.2
Markdown                          3.4.1
MarkupSafe                        2.1.1
matplotlib                        3.6.0
mlxtend                           0.19.0
more-itertools                    9.0.0
msgpack                           1.0.4
multi-rake                        0.0.2
murmurhash                        1.0.9
networkx                          2.8.8
ninja                             1.11.1
nltk                              3.8
numba                             0.56.0
numexpr                           2.8.7
numpy                             1.26.4
nvidia-cublas-cu12                12.5.3.2
nvidia-cuda-cupti-cu12            12.5.82
nvidia-cuda-nvrtc-cu12            12.5.82
nvidia-cuda-runtime-cu12          12.5.82
nvidia-cudnn-cu12                 9.2.0.82
nvidia-cufft-cu12                 11.2.3.61
nvidia-curand-cu12                10.3.6.82
nvidia-cusolver-cu12              11.6.3.83
nvidia-cusparse-cu12              12.5.1.3
nvidia-ml-py                      12.535.133
nvidia-ml-py3                     7.352.0
nvidia-nccl-cu12                  2.22.3
nvidia-nvtx-cu12                  12.5.82
oauthlib                          3.1.0
omegaconf                         2.0.6
openai                            1.0.0
opencv-python                     4.5.4.60
opencv-python-headless            4.5.4.58
openpyxl                          3.1.2
opt-einsum                        3.3.0
outlines                          0.0.43
packaging                         22.0
pandas                            1.3.5
partd                             1.2.0
pathy                             0.10.1
patsy                             0.5.3
pdf2image                         1.17.0
pdfminer.six                      20221105
pdfplumber                        0.8.0
PeakUtils                         1.3.0
phonenumbers                      8.12.55
Pillow                            9.4.0
pip                               22.0.4
pluggy                            0.6.0
portalocker                       2.6.0
portend                           3.1.0
preshed                           3.0.8
presidio-analyzer                 2.2.32
presidio-anonymizer               2.2.32
prometheus-client                 0.18.0
prometheus-fastapi-instrumentator 7.0.0
protobuf                          4.21.12
psutil                            5.9.4
py                                1.11.0
py-cpuinfo                        7.0.0
py-spy                            0.3.14
py4j                              0.10.9.5
pyarrow                           14.0.1
pyasn1                            0.4.8
pyasn1-modules                    0.2.8
pybind11                          2.10.3
pycld2                            0.41
pycocotools                       2.0.7
pycparser                         2.21
pycryptodome                      3.16.0
pydantic                          2.0
pyIsEmail                         2.0.1
pymongo                           4.3.3
PyMySQL                           0.7.6
pyod                              0.7.4
pyparsing                         2.4.5
PyPDF2                            3.0.1
pyrsistent                        0.19.3
pysbd                             0.3.3
pytest                            3.4.2
pytest-html                       1.16.0
pytest-metadata                   2.0.4
python-dateutil                   2.8.0
pytz                              2022.7
PyWavelets                        1.4.1
PyYAML                            6.0
ray                               2.9.0
regex                             2022.10.31
repoze.lru                        0.7
requests                          2.28.1
requests-file                     1.5.1
requests-oauthlib                 1.3.1
rsa                               4.0
s3fs                              0.4.2
s3transfer                        0.6.0
sacrebleu                         2.3.1
sacremoses                        0.0.53
safetensors                       0.4.1
scikit-image                      0.19.3
scikit-learn                      1.2.0
scipy                             1.9.3
sentence-transformers             2.7.0
sentencepiece                     0.1.96
setuptools                        58.1.0
shap                              0.41.0
six                               1.16.0
slicer                            0.0.7
smart-open                        6.3.0
sortedcontainers                  2.4.0
soupsieve                         2.0
spacy                             3.5.1
spacy-legacy                      3.0.11
srsly                             2.4.5
statsmodels                       0.13.2
tabulate                          0.9.0
tblib                             1.7.0
tempora                           5.2.0
tensorboard                       2.10.1
tensorboard-data-server           0.6.1
tensorboard-plugin-wit            1.6.0.post3
tensorflow                        2.16.2
tensorflow-estimator              2.15.0
tensorflow-hub                    0.16.1
tensorflow-io-gcs-filesystem      0.37.1
termcolor                         2.2.0
thinc                             8.1.12
threadpoolctl                     3.1.0
tifffile                          2022.10.10
tiktoken                          0.6.0
tldextract                        3.4.0
tokenizers                        0.19.1
toolz                             0.12.0
torch                             2.3.0
torchaudio                        2.3.0
torchvision                       0.18.0
tornado                           6.2
tqdm                              4.64.0
transformers                      4.40.1
tsfresh                           0.16.0
typer                             0.3.2
typing_extensions                 4.11.0
urllib3                           1.26.19
uvicorn                           0.23.2
vaderSentiment                    3.3.1
vllm                              0.5.0.post1
vllm-flash-attn                   2.5.9
Wand                              0.6.10
wasabi                            0.10.1
websocket-client                  1.4.2
Werkzeug                          2.2.2
wrapt                             1.14.1
xformers                          0.0.26.post1
xgboost                           0.81
xlrd                              2.0.1
zc.lockfile                       2.0
zict                              2.2.0
zipp                              3.11.0`

**Comment by ajayspatil7** [comment]
Can you try installing it this way `pip install --upgrade huggingface_hub`
Let Me Know if it works or not. :)

**Comment by nlpkiddo-2001** [comment]
@ajayspatil7 tried that, still getting the same issue

**Comment by Wauplin** [comment]
Hi @nlpkiddo-2001, sorry you're facing this issue. Are you using Python3.7? This could explain why you can't find a version above 0.17.1 since Python 3.7 support has been removed last year -and Python 3.8 end of support is supposed to happen [soon](https://devguide.python.org/versions/) as well-. For a correct installation, it is recommended to use Python 3.10 for instance (or even higher if your setup allows it).
